app/src/main/java/org/eyeseetea/malariacare/layout/adapters/survey/navigation/NavigationController.java

```diff
@@ -111,6 +111,9 @@ public boolean isNextAllowed() {
         OptionDB currentOptionDB = currentValueDB == null ? null : currentValueDB.getOptionDB();
         //Find next node with current option
         boolean isAllowed = findNext(currentOptionDB) != null;
+        if (!isAllowed && !hasNext(null) && !hasNext(currentValueDB.getOptionDB())) {
```

Comment:
@manuelplazaspalacio I think that we should not change this method of NavigationController for this because we are modifying a existed method that using several apps and several scenarios and is possible that we are changing a behavior. I think we should change caller (onSwipeLeft) for this method and create new condition to know if is the last question

Reply:
This method is only called from onSwipeLeft method of dyanmicTabAdapter. And I think if its the las question that it's the correct behaviour for all apps.